query-node/mappings/src/content/app.ts

```diff
@@ -0,0 +1,123 @@
+import { DatabaseManager, SubstrateEvent } from '@joystream/hydra-common'
+import { CreateApp, ICreateApp, IDeleteApp, IUpdateApp } from '@joystream/metadata-protobuf'
+import { MemberId } from '@joystream/types/primitives'
+import { App } from 'query-node/dist/model'
+import { logger } from '../common'
+
+export async function processCreateAppMessage(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  memberId: MemberId,
+  message: ICreateApp
+): Promise<void> {
+  const { name, appMetadata } = message
+  const appId = await getAppId(store, event)
+
+  const isAppExists = await store.get(App, {
+    where: {
+      name: name,
+    },
+  })
+
+  if (isAppExists) {
+    logger.error('App already exists', { name })
+    return
+  }
```

Comment:
For the parent issue App name should be: `name: immutable`. So it does not says anything about being unique so I think we should allow having apps with duplicate names, anyways please ask Beheho/Dmitry for clarity on this.

Reply:
Talked to Bedeho and we decided that the app name should be unique.